File: python/test/character_api_test.py
import os
import sys
import unittest

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../')))

from maplestory_openapi.api.kms import MapleStoryApi

api_key = os.environ['API_KEY']  # Your API Key
api = MapleStoryApi(api_key)

ocid = 'e0a4f439e53c369866b55297d2f5f4eb'

# 캐릭터 정보 조회

class CharacterApi(unittest.IsolatedAsyncioTestCase):
    async def test_character(self):
        nickname = '아델'
        response = await api.get_character_id(nickname)
        print(response)

    async def test_character_basic(self):
        response = await api.get_character_basic(ocid)
        print(response)

    async def test_character_image(self):
        response = await api.get_character_image(ocid)
        print(response)

    async def test_character_popularity(self):
        response = await api.get_character_popularity(ocid)
        print(response)

    async def test_character_stat(self):
        response = await api.get_character_stat(ocid)
        print(response)

    async def test_character_hyper_stat(self):
        response = await api.get_character_hyper_stat(ocid)
        print(response)

    async def test_character_propensity(self):
        response = await api.get_character_propensity(ocid)
        print(response)

    async def test_character_ability(self):
        response = await api.get_character_ability(ocid)
        print(response)

    async def test_character_item_equipment(self):
        response = await api.get_character_item_equipment(ocid)
        print(response)

    async def test_character_cashitem_equipment(self):
        response = await api.get_character_cashitem_equipment(ocid)
        print(response)

    async def test_character_symbol_equipment(self):
        response = await api.get_character_symbol_equipment(ocid)
        print(response)

    async def test_character_set_effect(self):
        response = await api.get_character_set_effect(ocid)
        print(response)

    async def test_character_beauty_equipment(self):
        response = await api.get_character_beauty_equipment(ocid)
        print(response)

    async def test_character_android_equipment(self):
        response = await api.get_character_android_equipment(ocid)
        print(response)

    async def test_character_pet_equipment(self):
        response = await api.get_character_pet_equipment(ocid)
        print(response)

    async def test_character_skill(self):
        skill_grade = '0'
        response = await api.get_character_skill(ocid, skill_grade)
        print(response)

    async def test_character_link_skill(self):
        response = await api.get_character_link_skill(ocid)
        print(response)

    async def test_character_v_matrix(self):
        response = await api.get_character_vmatrix(ocid)
        print(response)

    async def test_character_hexa_matrix(self):
        response = await api.get_character_hexamatrix(ocid)
        print(response)

    async def test_character_hexa_matrix_stat(self):
        response = await api.get_character_hexamatrix_stat(ocid)
        print(response)

    async def test_character_dojang(self):
        response = await api.get_character_dojang(ocid)
        print(response)

    async def test_character_other_stat(self):
        response = await api.get_character_other_stat(ocid)
        print(response)

    async def test_character_ring_exchange_skill_equipment(self):
        response = await api.get_character_ring_exchange_skill_equipment(ocid)
        print(response)


if __name__ == '__main__':
    unittest.main()

File: js/test/characterApi.test.ts
import * as process from 'process';

import { toString } from './utils';
import { MapleStoryApi } from '../src/maplestory/api/kms/mapleStoryApi';

const apiKey = process.env.API_KEY!; // Your API Key
const api = new MapleStoryApi(apiKey);

const ocid = 'e0a4f439e53c369866b55297d2f5f4eb';

describe('캐릭터 정보 조회', () => {
  test('캐릭터 식별자(ocid) 조회', async () => {
    const nickname = '아델';
    const response = await api.getCharacter(nickname);

    console.log(toString(response));
  });

  test('캐릭터 목록 조회', async () => {
    const response = await api.getCharacterList();

    console.log(toString(response));
  });

  test('기본 정보 조회', async () => {
    const response = await api.getCharacterBasic(ocid);

    console.log(toString(response));
  });

  test('외형 이미지 조회', async () => {
    const response = await api.getCharacterImage(ocid);

    console.log(toString(response));
  });

  test('인기도 정보 조회', async () => {
    const response = await api.getCharacterPopularity(ocid);

    console.log(toString(response));
  });

  test('종합 능력치 정보 조회', async () => {
    const response = await api.getCharacterStat(ocid);

    console.log(toString(response));
  });

  test('하이퍼스탯 정보 조회', async () => {
    const response = await api.getCharacterHyperStat(ocid);

    console.log(toString(response));
  });

  test('성향 정보 조회', async () => {
    const response = await api.getCharacterPropensity(ocid);

    console.log(toString(response));
  });

  test('어빌리티 정보 조회', async () => {
    const response = await api.getCharacterAbility(ocid);

    console.log(toString(response));
  });

  test('장착 장비 정보 조회 (캐시 장비 제외)', async () => {
    const response = await api.getCharacterItemEquipment(ocid);

    console.log(toString(response));
  });

  test('장착 캐시 장비 정보 조회', async () => {
    const response = await api.getCharacterCashItemEquipment(ocid);

    console.log(toString(response));
  });

  test('장착 심볼 정보 조회', async () => {
    const response = await api.getCharacterSymbolEquipment(ocid);

    console.log(toString(response));
  });

  test('적용 세트 효과 정보 조회', async () => {
    const response = await api.getCharacterSetEffect(ocid);

    console.log(toString(response));
  });

  test('장착 헤어, 성형, 피부 정보 조회', async () => {
    const response = await api.getCharacterBeautyEquipment(ocid);

    console.log(toString(response));
  });

  test('장착 안드로이드 정보 조회', async () => {
    const response = await api.getCharacterAndroidEquipment(ocid);

    console.log(toString(response));
  });

  test('장착 펫 정보 조회', async () => {
    const response = await api.getCharacterPetEquipment(ocid);

    console.log(toString(response));
  });

  test('스킬 정보 조회', async () => {
    const skillGrade = '6';
    const response = await api.getCharacterSkill(ocid, skillGrade);

    console.log(toString(response));
  });

  test('장착 링크 스킬 정보 조회', async () => {
    const response = await api.getCharacterLinkSkill(ocid);

    console.log(toString(response));
  });

  test('V매트릭스 정보 조회', async () => {
    const response = await api.getCharacterVMatrix(ocid);

    console.log(toString(response));
  });

  test('HEXA 코어 정보 조회', async () => {
    const response = await api.getCharacterHexaMatrix(ocid);

    console.log(toString(response));
  });

  test('HEXA 매트릭스 설정 HEXA 스탯 정보 조회', async () => {
    const response = await api.getCharacterHexaMatrixStat(ocid);

    console.log(toString(response));
  });

  test('무릉도장 최고 기록 정보 조회', async () => {
    const response = await api.getCharacterDojang(ocid);

    console.log(toString(response));
  });
});
